src/transformers/trainer_utils.py

```diff
@@ -622,3 +622,49 @@ class FSDPOption(ExplicitEnum):
     SHARD_GRAD_OP = "shard_grad_op"
     OFFLOAD = "offload"
     AUTO_WRAP = "auto_wrap"
+
+
+UNUSED_COLUMNS_LOGGING_MESSAGE = (
+    "The following columns {dset_description} don't have a corresponding argument in "
+    "`{model_name}.forward` and have been ignored: {ignored_columns}."
+    " If {ignored_columns} are not expected by `{model_name}.forward`, "
+    " you can safely ignore this message."
+)
```

Comment:
As said before, let's leave the warning to where it's used, no need to refactor it in a variable.